CONTRIBUTING.md

```diff
@@ -43,6 +43,21 @@ $(which git && git --version)
 
 We love contributions to Realm! If you'd like to contribute code, documentation, or any other improvements, please [file a Pull Request](https://github.com/realm/realm-cocoa/pulls) on our GitHub repository. Make sure to accept our [CLA](#CLA) and to follow our [style guide](https://github.com/realm/realm-cocoa/wiki/Objective-C-Style-Guide).
 
+### Commit Messages
+
+Although we don’t enforce a strict format for commit messages, we prefer that you follow the guidelines below, which are common among open source projects. Following these guidelines helps with the review process, searching commit logs and documentation of implementation details. At a high level, the contents of the commit message should convey the rationale of the change, without delving into much detail. For example, `setter names were not set right` leaves the reviewer wondering about which bits and why they weren’t “right”. In contrast, `[RLMProperty] Correctly capitalize setterName` conveys almost all there is to the change.
+
+Below are some guidelines about the format of the commit message itself:
+
+* Separate the commit message into a single-line title and a separate body that describes the change.
+* Make the title concise to be easily read within a commit log.
+* In changes that are restricted to a specific part of the code, include a [tag] at the start of the line in square brackets — for example, `[RLMArray] …`, `[scripts] …` or  `[Podspec] …`. This tag helps email filters and searches for post-commit reviews.
```

Comment:
@mrackwitz can you please just remove this point from the document since it's the only part that's been disputed? That way we can get the rest of the document in, and take the debate about tags to Slack and move on.